core/src/main/scala/kafka/server/metadata/ZkMetadataCache.scala

```diff
@@ -355,6 +375,13 @@ class ZkMetadataCache(brokerId: Int, metadataVersion: MetadataVersion, brokerFea
         case id if id < 0 => None
         case id => Some(id)
       }
+      val kraftControllerIdOpt = updateMetadataRequest.kraftControllerId() match {
+        case id if id < 0 => None
+        case id => Some(id)
+      }
+      if (controllerIdOpt.nonEmpty && kraftControllerIdOpt.nonEmpty) {
```

Comment:
This isn't an error, right? It's possible for both of them to be -1... Like sometimes we are in the middle of electing a new controller.

Reply:
Removed this logic in favor of new change to the RPC message.